Assets/Scripts/GameElements/PlayerController.cs

```diff
@@ -209,8 +209,18 @@ protected void OnTriggerEnter2D(Collider2D other)
         {
             eggController = other.gameObject.GetComponent<EggController>();
         }
-    }
 
+        // TODO: ideally we handle this better with a script we can attach to any game object separate from main logic
+        // TODO: something like a damage emitter and damage accumulator script or whatever and that can handle checking and storing
+        // TODO: damage on its own
```

Comment:
Why not have all enemy obstacle object controllers extend a base abstract class that has an `OnCollision(PlayerController player)` function?

Reply:
Oh note this isn't a PR blocking comment, just an idea for the future!

Reply:
oooo I like it!
I think creating damage emitters and accumulators would be more generic and would allow for future types to hurt things other than the player more seamlessly too
But I like this idea
It is a real struggle deciding what should handle collisions lmao